diff --git a/build/envsetup.sh b/build/envsetup.sh
index f2de7502..582d46be 100644
--- a/build/envsetup.sh
+++ b/build/envsetup.sh
@@ -12,8 +12,8 @@ function check_product()
         echo "Couldn't locate the top of the tree. Try setting TOP." >&2
         return
     fi
-    if (echo -n $1 | grep -q -e "^lineage_") ; then
-        LINEAGE_BUILD=$(echo -n $1 | sed -e 's/^lineage_//g')
+    if (echo -n $1 | grep -q -e "^lineage_" -e "^quorin_") ; then
+        LINEAGE_BUILD=$(echo -n $1 | sed -e 's/^lineage_//g' -e 's/^quorin_//g')
     else
         LINEAGE_BUILD=
     fi
